core/templates/dev/head/profile/PreferencesSpec.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('Preferences Controller', function() {
+  describe('PreferencesCtrl', function() {
+    var scope, ctrl, $Http;
+
+    beforeEach(function() {
+      module('oppia');
+    });
+
+    beforeEach(inject(function(_$httpBackend_, $http, $rootScope, $controller) {
+      _$httpBackend_.expectGET('/signuphandler/data').respond({
+        username: 'myUsername',
+        has_agreed_to_latest_terms: true
+      });
+
+      $Http = $http;
+
+      mockAlertsService = {};
+
+      scope = $rootScope.$new();
+
+      ctrl = $controller('Preferences', {
+        $scope: scope,
+        $http: $http,
+        $rootScope: $rootScope,
+        alertsService: mockAlertsService
+      });
+    }));
+
+    it('should show that editor role notifications checkbox is true by default',
+      function() {
+        $Http.get(scope._PREFERENCES_DATA_URL).then(function(response) {
```

Comment:
Generally, in tests, the implementation should not be copied. Otherwise, if the code changes, the tests are testing the wrong thing.